mne/channels/tests/test_channels.py

```diff
@@ -71,10 +71,12 @@ def test_set_channel_types():
     # Test change to illegal channel type
     mapping = {'EOG 061': 'xxx'}
     assert_raises(ValueError, raw.set_channel_types, mapping)
+    # Test changing type if in proj (avg eeg ref here)
+    mapping = {'EEG 060': 'eog', 'EEG 059': 'ecg', 'EOG 061': 'seeg'}
+    assert_raises(RuntimeError, raw.set_channel_types, mapping)
     # Test type change
-    raw2 = Raw(raw_fname)
+    raw2 = Raw(raw_fname, add_eeg_ref=False)
```

Comment:
so will this throw an error if `proj` has not been applied but is added and the user is trying to change the channel type?